cmd/api/auth/authz.go

```diff
@@ -0,0 +1,51 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"fmt"
+	"github.com/gin-gonic/gin"
+	authzv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientAuthzv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
```

Comment:
The Z is because there is also the client for authentication which I will be using when I implement the middleware for the authN. See [authentication/v1](https://pkg.go.dev/k8s.io/client-go@v0.30.0/kubernetes/typed/authentication/v1)

The v1 is because the v1beta1 could be used, see [authorization/](https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/authorization@v0.30.0)

Note that I'm also following the way this are used in the library. See [metav1 import](https://github.com/kubernetes/client-go/blob/v0.30.0/kubernetes/typed/authorization/v1/subjectaccessreview.go#L25).

It's also how things are imported in the [example](https://github.com/kubernetes/client-go/blob/v0.30.0/examples/in-cluster-client-configuration/main.go) of the library.

Reply:
I'll solve this but feel free to reopen it even if the PR is merged, I am open to do adjustments.